python/ray/experimental/channel/common.py

```diff
@@ -355,8 +363,11 @@ def close(self):
 
 @DeveloperAPI
 class WriterInterface:
-    def __init__(self, output_channel: ChannelInterface):
-        self._output_channel = output_channel
+    def __init__(
+        self, output_channels: List[ChannelInterface], output_idxs: List[Optional[int]]
```

Comment:
why is it optional int? can you update a docstring for this interface?

Reply:
Done.
For consistency, I assert len(output_channels) == len(output_idxs).
Thus, when there is only one output channel, it means the whole value is returned, in this case I let output_idxs = [None], namely returning the whole value when the index is None.